### PR TITLE
added initial set of descriptions for VM and VMSS properties

### PR DESCRIPTION
**Pull request by gatneil**

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json [hunk 1]
```diff
@@ -3413,11 +3413,11 @@
         },
         "type": {
           "type": "string",
```
Comment (veronicagg) [inline specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json]: based on the description you may want to evaluate turning this property into an enum in the future - https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-enum

File: specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json [hunk 2]
```diff
@@ -4893,7 +4893,7 @@
...
-          "description": "The type of identity used for the virtual machine scale set. Currently, the only supported type is 'SystemAssigned', which implicitly creates an identity.",
+          "description": "The type of indentity used for the virtual machine. Currently, the only supported type is 'SystemAssigned', which implicitly creates an identity.",
```
Comment (veronicagg) [inline specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json]: typo: indentity -> identity?

File: specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json [hunk 3]
```diff
@@ -4976,14 +4976,14 @@
...
-      "description" : "Describes The SKU capabilites object."
+      "description": "Describes The SKU capabilites object."
     },
     "ResourceSkuRestrictions": {
       "properties": {
         "type": {
           "type": "string",
           "readOnly": true,
-          "description": "The type of restrictions.",
+          "description": "The type of restrictions",
```
Comment (veronicagg) [inline specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json]: "." at the end?

File: specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json [hunk 4]
```diff
@@ -5216,20 +5216,20 @@
...
         },
         "caching": {
           "$ref": "#/definitions/Caching",
-          "description": "The caching type."
+          "description": "Specifies the caching requirements. <br><br> Possible values are: <br><br> **None** <br><br> **ReadOnly** <br><br> **ReadWrite** <br><br> Default: **None for Standard storage. ReadOnly for Premium storage**"
         },
         "createOption": {
           "$ref": "#/definitions/CreateOption",
-          "description": "The create option."
+          "description": "The create option"
```
Comment (veronicagg) [inline specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json]: "." at the end?

File: specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json [hunk 5]
```diff
@@ -5472,23 +5472,23 @@
         },
         "type": {
           "type": "string",
-          "description": "The type of the extension handler."
+          "description": "Specifies the type of the extension; an example is \"CustomScriptExtension\""
```
Comment (veronicagg) [inline specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json]: "." at the end?

File: specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json [hunk 6]
```diff
@@ -5551,27 +5551,27 @@
...
         },
         "networkProfile": {
           "$ref": "#/definitions/VirtualMachineScaleSetNetworkProfile",
-          "description": "The virtual machine scale set network profile."
+          "description": "Specifies properties of the network interfaces of the virtual machines in the scale set."
         },
         "diagnosticsProfile": {
           "$ref": "#/definitions/DiagnosticsProfile",
-          "description": "The virtual machine scale set diagnostics profile."
+          "description": "Specifies the boot diagnostic settings state. <br><br>Minimum api-version: 2015-06-15"
```
Comment (veronicagg) [inline specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json]: "." at the end?

**Review by veronicagg — CHANGES_REQUESTED** [review]
Few minor comments, you may also want to evaluate adding missing descriptions flagged by linter tool, example: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/267008359#L723, https://travis-ci.org/Azure/azure-rest-api-specs/jobs/267008359#L1695

File: specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json (hunk 1 above)
Comment (gatneil) [inline specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json]: Unfortunately the list of extension types is not static or versioned (at least, not in any user-specifiable way); so I don't think an enum would work for it (I *wish* it were versioned :))

File: specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json (hunk 2 above)
Comment (gatneil) [inline specification/compute/resource-manager/Microsoft.Compute/2017-03-30/compute.json]: lol; thanks

**Comment by gatneil** [comment]
@veronicagg thanks for the review! Addressed the comments above. This particular PR is meant to get a jump start on descriptions, but look forward to more incremental follow-up PRs over time to bring it up to 100% description coverage :)

**Review by veronicagg — APPROVED** [review]

**Pull request merged**